api/v1/controllers/EventController.js

```diff
@@ -0,0 +1,150 @@
+var bodyParser = require('body-parser');
+var _Promise = require('bluebird');
+
+var errors = require('../errors');
+var services = require('../services');
+var config = require('../../config');
+
+var middleware = require('../middleware');
+var requests = require('../requests');
+var utils = require('../utils');
+var roles = require('../utils/roles');
+
+var EventService = require('../services/EventService');
+
+var logger = require('../../logging');
+
+var router = require('express').Router();
+
+function getEvents(req, res, next) {
+        var updated = req.query.lastUpdated || 0;
+        return services.EventService.findByUpdated(updated)
+                .then(function(result) {
+                        res.body = result.length == 0 ? [] : result.map(function (obj) {
+                                return obj.toJSON();
+                        });
+                        next();
+                        return null;
+                });
+}
+
+function createEvent(req, res, next) {
+        return services.EventService.createEvent(req.body.name,
+                                                 req.body.shortName,
+                                                 req.body.description,
+                                                 req.body.qrCode,
+                                                 req.body.startTime,
+                                                 req.body.endTime,
+                                                 req.body.tag || "SCHEDULE",
+                                                 req.body.locations || [])
+                .then(function(result) {
+                        res.body = result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function createLocation(req, res, next) {
+        return services.EventService.createLocation(req.body.name,
+                                                    req.body.shortName,
+                                                    req.body.latitude,
+                                                    req.body.longitude)
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function getLocations(req, res, next) {
+        return services.EventService.getLocations()
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function addLocationToEvents(req, res, next) {
```

Comment:
What's the exact request body here? My understanding is that it adds a location to a group of events, so shouldn't it return the updated events as the response?